bftengine/src/bftengine/Digest.cpp

```diff
@@ -102,7 +102,7 @@ std::string Digest::toString() const {
   for (size_t i = 0; i < DIGEST_SIZE; i++) {
     // TODO(DD): Is it by design?
     // NOLINTNEXTLINE(bugprone-signed-char-misuse)
-    unsigned int b = (unsigned int)d[i];
+    unsigned int b = (unsigned char)d[i];
```

Comment:
This is bug i found while testing my changes. Digest string does not print correctly with the existing code. Here, we are casting signed char to unsigned int which is bug prone. For more details, please check this link
https://releases.llvm.org/10.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/bugprone-signed-char-misuse.html

Reply:
I think this should be changed in a separate PR if it's a bug since this is not related to checkpoint.
As b is unsigned int, the conversion from signed char to unsigned is already happening.

Reply:
Actually, this is used in the integrity check code i added for db  checkpoint. 
Here, the problem as stated in link shared above: 
_The basic problem with the signed char, that it might store the non-ASCII characters as negative values. The human programmer probably expects that after an integer conversion the converted value matches with the character code (a value from [0..255]), however, the actual value is in [-128..127] interval._
_To avoid this kind of misinterpretation, the desired way of converting from a signed char to an integer value is converting to unsigned char first, which stores all the characters in the positive [0..255] interval which matches with the known character codes._

I can raise a separate jira for this bug and link the PR there.